### PR TITLE
rough flake8 integration

### PR DESCRIPTION
**Pull request by collinanderson**

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -7,5 +7,6 @@ alabaster>=0.6.0
 nose==1.3.0
 spec>=0.11.1
 mock==1.0.1
+flake8
```
Comment (bitprophet) [inline dev-requirements.txt]: Can you stick the latest stable release number in here as an `==`? thanks!

File: invoke/monkey.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Fuckin' A.
 
-import select, errno, os, sys
+import errno
```
Comment (bitprophet) [inline invoke/monkey.py]: Unclear why this file is being touched, it's gone in HEAD. Can you try rebasing on latest HEAD perhaps?

File: invoke/parser/parser.py [hunk 3]
```diff
@@ -88,7 +88,7 @@ def parse_argv(self, argv):
             # flag value and still in valid parsing territory (i.e. not in
             # "unknown" state which implies store-only)
             if not machine.waiting_for_flag_value and is_flag(token) \
-                and not machine.result.unparsed:
+                    and not machine.result.unparsed:
```
Comment (bitprophet) [inline invoke/parser/parser.py]: Is 8-space indents on continuations a flake8 thing? It's not my preferred style :( (Honestly, I thought flake8 would complain about backslash continuations entirely, parentheticals are better anyway).

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+exclude=invoke/vendor,sites,tests
```
Comment (bitprophet) [inline setup.cfg]: Any particular reason to exclude `tests`?

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+ignore=E124,E125,E128,E261,E302,E402,F401,W503,E712,F821
```
Comment (bitprophet) [inline setup.cfg]: I should look these up to see what they are, am not a linting expert.

**Comment by bitprophet** [comment]
Thanks for poking at this! See line comments :)

**Comment by bitprophet** [comment]
Also linking to #157 which this kinda supersedes.

File: invoke/parser/parser.py (hunk 3 above)
Comment (collinanderson) [inline invoke/parser/parser.py]: It just doesn't want the conditional indented at the same level as the code.

File: setup.cfg (hunk 5 above)
Comment (collinanderson) [inline setup.cfg]: Yeah, I figure start with a lot of them ignored. If individual ones get fixed they can be removed from this list.

File: invoke/parser/parser.py (hunk 3 above)
Comment (collinanderson) [inline invoke/parser/parser.py]: E129 visually indented line with same indent as next logical line

File: dev-requirements.txt (hunk 1 above)
Comment (collinanderson) [inline dev-requirements.txt]: done

File: invoke/monkey.py (hunk 2 above)
Comment (collinanderson) [inline invoke/monkey.py]: Ahh. Yup.

File: invoke/parser/parser.py (hunk 3 above)
Comment (collinanderson) [inline invoke/parser/parser.py]: This is now ignored under E125 for now.

File: setup.cfg (hunk 4 above)
Comment (collinanderson) [inline setup.cfg]: I force-pushed a change that now includes tests too. I just didn't want the diff to be too big :)

File: setup.cfg (hunk 5 above)
Comment (collinanderson) [inline setup.cfg]: Basically, you can remove one from the list, then run flake8 to see what lines they apply to.

**Comment by bitprophet** [comment]
Thanks for making those updates. Annoyingly, I didn't notice last time that there's apparently a rule for double spaces before line comments (e.g. `foo = blah  # lol`) - this is another spot where I personally diverge from core PEP8.

Went and checked on all the stuff ignored/not ignored to see if I agree (I'd like to choose a single set of these rule ignores so I can apply them eventually to all of my projects :))...results:

* As above, I'm thinking we ignore `E261` (two spaces before line comments).
* Why ignore `F401` (module imported but unused)? When I run w/ it un-ignored I see 90% correct catches of unused imports (boy I should've put linting in before! thanks again for the PR...), with a handful of `__init__.py` convenience imports that we can presumably explicitly ignore.
* I don't see `W503` in pep8's docs, what's it do?
* Where are we violating `E711` and `E712`? Offhand I'd agree with those errors if they popped up. If I remove those from a copy of your branch, I see a handful of alerts and they all seem legit (where we use `==` instead of the arguably more correct `is`).
* `F811` and `F821` also sound like rules I'd agree with if they alerted, and as above, if I remove them it does find some things I would consider errors :)

**Comment by collinanderson** [comment]
Don't worry, I added two space comments in the second change which is why you didn't catch it the first time.

I originally ignored all of the import cleanup just to keep the diff small enough. Here it is with all of the import cleanup.

**Comment by bitprophet** [comment]
Looks good, thanks again! I might tweak that ignores list more in future but this is a great start.

**Pull request merged**